catalog/core/catalog-core-encryption/src/main/java/org/codice/ddf/catalog/content/encryption/ContentEncryptorDecryptor.java

```diff
@@ -0,0 +1,275 @@
+  private static final String CONTENT_ENCRYPTION_KEY_ALIAS_VALUE =
+      SecurityConstants.getContentEncryptionKeyAlias();
+  private static final String CONTENT_ENCRYPTION_KEY_PASSWORD_VALUE =
+      SecurityConstants.getContentEncryptionKeyPassword();
+  private static final String AES = "AES";
+  private static final String CIPHER_TYPE = "AES/GCM/NoPadding";
+  private static final String ENABLE_DEBUG_LOGGING_MSG =
+      " Enable debug logging for more information.";
+  private static final int KEY_LENGTH = 256;
+  @VisibleForTesting static final int IV_BYTE_SIZE = 12;
+  @VisibleForTesting static final int AUTH_TAG_LENGTH = 128;
+
+  private static SecureRandom secureRandom = new SecureRandom();
+  private static Cipher cipher;
+  private static KeyStore keyStore;
+  private static KeyStore trustStore;
+  private static SecretKey contentEncryptionKey;
+
+  public ContentEncryptorDecryptor() {
+    initKeystores();
+    initKey();
+    initCipher();
+  }
+
+  @VisibleForTesting
+  ContentEncryptorDecryptor(SecretKey secretKey) {
+    contentEncryptionKey = secretKey;
+    initCipher();
+  }
+
+  @Override
+  public byte[] encrypt(InputStream inputStream) throws ContentEncryptionException {
+    try {
+      return encrypt(ByteStreams.toByteArray(inputStream));
+    } catch (IOException e) {
+      LOGGER.warn("Problem reading input stream.");
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] encrypt(byte[] inputBytes) throws ContentEncryptionException {
+    try {
+      readyCipherForEncryption();
+      return wrapWithIV(cipher.getIV(), cipher.doFinal(inputBytes));
+    } catch (BadPaddingException | IllegalBlockSizeException e) {
+      LOGGER.warn("Problem encrypting file content." + ENABLE_DEBUG_LOGGING_MSG);
+      LOGGER.debug("", e);
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] decrypt(InputStream inputStream) throws ContentEncryptionException {
+    try {
+      return decrypt(ByteStreams.toByteArray(inputStream));
+    } catch (IOException e) {
+      LOGGER.warn("Problem reading input stream.");
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] decrypt(byte[] inputBytes) throws ContentEncryptionException {
+    try {
+      // separate the iv from the encrypted payload
+      ByteBuffer byteBuffer = ByteBuffer.wrap(inputBytes);
+      byte[] iv = new byte[IV_BYTE_SIZE];
+      byte[] encryptedBytes = new byte[inputBytes.length - IV_BYTE_SIZE];
+      byteBuffer.get(iv);
+      byteBuffer.get(encryptedBytes);
+
+      readyCipherForDecryption(iv);
+      return cipher.doFinal(encryptedBytes);
+    } catch (BadPaddingException | IllegalBlockSizeException e) {
+      LOGGER.warn("Problem decrypting file content." + ENABLE_DEBUG_LOGGING_MSG);
+      LOGGER.debug("", e);
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  private static void initKeystores() {
```

Comment:
📘 Copied and modified from https://github.com/codice/ddf/blob/76bf9aa553efd398e27375216ca7af9b4fdcc6d2/platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java#L127